packages/server-wallet/src/wallet/index.ts

```diff
@@ -99,8 +100,25 @@ export class Wallet implements WalletInterface {
   async joinChannel(_args: JoinChannelParams): Result {
     throw 'Unimplemented';
   }
-  async updateChannel(_args: UpdateChannelParams): Result {
-    throw 'Unimplemented';
+  async updateChannel({channelId, appData, allocations}: UpdateChannelParams): Result {
+    return Channel.transaction(async tx => {
+      const channel = await Store.getChannel(channelId, undefined);
+
+      const outcome = deserializeAllocations(allocations);
+      const decision = updateChannel({channelId, appData, outcome}, channel);
+
+      switch (decision._tag) {
```

Comment:
https://gcanti.github.io/fp-ts/modules/Either.ts.html#isleft

Would this be better than inspecting an internal property?